mono/metadata/marshal.c

```diff
@@ -737,7 +737,7 @@ mono_array_to_lparray (MonoArray *array)
 		nativeArraySize = array->max_length;
 		nativeArray = (void **)malloc(sizeof(gpointer) * nativeArraySize);
```

Comment:
You didn't change this, but I see we are mixing `malloc` with `g_free`. Should probably use `g_malloc`.